osbuild/util/rhsm.py

```diff
@@ -0,0 +1,54 @@
+"""Red Hat Subscription Manager support module
+
+This module implements utilities that help with interactions
+with the subscriptions attached to the host machine.
+"""
+
+import configparser
+import re
+
+
+class Subscriptions:
+    def __init__(self, repositories):
+        self.repositories = repositories
+
+    @classmethod
+    def from_host_system(cls):
+        with open('/etc/yum.repos.d/redhat.repo', 'r') as fp:
+            return cls.parse_repo_file(fp)
+
+    @staticmethod
+    def _process_baseurl(input_url):
+        # First escape meta characters that might occur in a URL
+        input_url = re.escape(input_url)
+
+        # Now replace variables with regexes (see man 5 yum.conf for the list)
+        for variable in ['\\$releasever', '\\$arch', '\\$basearch', '\\$uuid']:
+            input_url = input_url.replace(variable, '.*')
+
+        return re.compile(input_url)
+
+    @classmethod
+    def parse_repo_file(cls, fp):
+        parser = configparser.ConfigParser()
+        parser.read_file(fp)
+
+        repositories = dict()
+        for section in parser.sections():
+            repositories[section]=dict()
+            repositories[section]['baseurl']=cls._process_baseurl(parser.get(section, 'baseurl'))
```

Comment:
nit: the spacing around "=" is inconsistent. Maybe just use `autopep8` and be done with it manual formatting? ;)

Reply:
One other thing: maybe we should call the `baseurl` here `matchurl` or something so that we don't confuse the two if in the future anyone wants to reuse this class for something?